runtime/compiler/control/CompilationThread.cpp

```diff
@@ -6781,16 +6768,17 @@ bool isMethodIneligibleForAot(J9Method *method)
 bool
 TR::CompilationInfoPerThreadBase::shouldPerformLocalComp(const TR_MethodToBeCompiled *entry)
    {
-      bool doLocalComp = false;
-      static char *localColdCompilations = feGetEnv("TR_LocalColdCompilations");
-      // As a heuristic, cold compilations should be performed locally because
-      // they are supposed to be cheap with respect to memory and CPU.
-      //
-      if (entry->_optimizationPlan->getOptLevel() <= cold &&
-         (TR::Options::getCmdLineOptions()->getOption(TR_EnableJITaaSHeuristics) || localColdCompilations))
-         doLocalComp = true;
+   bool doLocalComp = false;
+   static char *localColdCompilations = feGetEnv("TR_LocalColdCompilations");
+   // As a heuristic, cold compilations should be performed locally because
+   // they are supposed to be cheap with respect to memory and CPU.
+   //
+   if (entry->_optimizationPlan->getOptLevel() <= cold &&
+      (TR::Options::getCmdLineOptions()->getOption(TR_EnableJITaaSHeuristics) || localColdCompilations) || 
+      !JITaaS::J9ClientStream::isServerCompatible(OMRPORT_FROM_J9PORT(_jitConfig->javaVM->portLibrary)))
```

Comment:
Don't you have to pass just the portLibrary here? `_jitConfig=>javaVM->portLibrary` instead of `OMR_PORT_FROM_J9PORT...`

Reply:
I thought we have a `J9PortLibrary` and a `OMRPortLibrary`
and _jitConfig->javaVM->portLibrary gives you the J9PortLibrary
and to get OMRPortLibrary, we need to use `OMRPORT_FROM_J9PORT`. Correct me if I'm wrong.

Reply:
You may be right. In that case I am ok with the code.